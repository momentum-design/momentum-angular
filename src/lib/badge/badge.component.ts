/** @component badge */
import { Component, ElementRef, Input } from '@angular/core';

@Component({
  selector: 'md-badge',
  template: `
    <ng-content></ng-content>
  `,
  host: {
    class: 'md-badge',
    '[class.md-badge--round]': 'rounded',
  },
})
export class BadgeComponent {
  /** @option Sets the optional rounded class | false */
  @Input() public rounded: boolean = false;

  private _color: string = null;
  /** @option Sets optional button color | null */
  @Input()
  set color(color: string) {
    if (this._color) {
      this.elementRef.nativeElement.classList.remove(
        `md-badge--${this._color}`
      );
    }
    this.elementRef.nativeElement.classList.add(`md-badge--${color}`);
    this._color = color;
  }

  constructor(private elementRef: ElementRef) {}
}
