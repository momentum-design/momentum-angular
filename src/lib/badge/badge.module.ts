import { NgModule, ModuleWithProviders } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BadgeComponent } from './badge.component';

@NgModule({
  imports: [CommonModule],
  declarations: [BadgeComponent],
  exports: [BadgeComponent],
})
export class BadgeModule {
  static forRoot(): ModuleWithProviders {
    return {
      ngModule: BadgeModule,
      providers: [],
    };
  }
}
