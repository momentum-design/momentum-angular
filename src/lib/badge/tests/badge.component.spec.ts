import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { Component } from '@angular/core';
import { BadgeComponent } from '../badge.component';

@Component({
  template: `
    <md-badge
      [ngClass]="'first second'"
      class="third fourth"
      color="blue"
      [rounded]="true"
      >Testing Classes</md-badge
    >
  `,
})
class TestHostComponent {}

describe('BadgeComponent', () => {
  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [BadgeComponent, TestHostComponent],
    }).compileComponents();
  }));

  describe('Default, Rounded and Colors', () => {
    let testComponent: BadgeComponent;
    let fixture: ComponentFixture<BadgeComponent>;
    let badgeNativeElement: HTMLElement;

    beforeEach(() => {
      fixture = TestBed.createComponent(BadgeComponent);
      testComponent = fixture.componentInstance;
    });

    it('should create and match snapshot', () => {
      fixture.detectChanges();
      expect(testComponent).toBeTruthy();
      expect(fixture).toMatchSnapshot();
    });

    it('should render one md-badge element with default class name', () => {
      fixture.detectChanges();

      badgeNativeElement = fixture.nativeElement;
      expect(badgeNativeElement.className).toEqual('md-badge');
    });

    it('should assign conditional rounded class name', () => {
      testComponent.rounded = true;
      fixture.detectChanges();
      expect(fixture).toMatchSnapshot();

      badgeNativeElement = fixture.nativeElement;
      expect(badgeNativeElement.className).toContain('md-badge--round');
    });

    it('should not assign conditional rounded class name', () => {
      testComponent.rounded = false;
      fixture.detectChanges();

      badgeNativeElement = fixture.nativeElement;
      expect(badgeNativeElement.className).not.toContain('md-badge--round');
    });

    it('should assign conditional color class name', () => {
      testComponent.color = 'blue';
      fixture.detectChanges();
      expect(fixture).toMatchSnapshot();

      badgeNativeElement = fixture.nativeElement;
      expect(badgeNativeElement.className).toContain('md-badge--blue');
    });

    it('should update conditional color class name if color is updated', () => {
      testComponent.color = 'red';
      fixture.detectChanges();

      badgeNativeElement = fixture.nativeElement;
      expect(badgeNativeElement.className).toContain('md-badge--red');

      testComponent.color = 'blue';
      fixture.detectChanges();

      expect(badgeNativeElement.className).not.toContain('md-badge--red');
      expect(badgeNativeElement.className).toContain('md-badge--blue');
    });

    it('should not assign conditional color class name', () => {
      fixture.detectChanges();

      badgeNativeElement = fixture.nativeElement;
      expect(badgeNativeElement.className).not.toContain('md-badge--blue');
    });

    it('should not assign conditional color class name', () => {
      fixture.detectChanges();

      badgeNativeElement = fixture.nativeElement;
      expect(badgeNativeElement.className).not.toContain('md-badge--blue');
    });
  });

  describe('Classes and ngClass', () => {
    let testHost: TestHostComponent;
    let fixture: ComponentFixture<TestHostComponent>;

    it('should pass classNames to the component', () => {
      fixture = TestBed.createComponent(TestHostComponent);
      testHost = fixture.componentInstance;
      const badge = fixture.nativeElement.querySelector('md-badge');
      fixture.detectChanges();
      expect(fixture).toMatchSnapshot();

      expect(badge.className).toContain('md-badge--blue');
      expect(badge.className).toContain('md-badge--round');
      expect(badge.className).toContain('first second');
      expect(badge.className).toContain('third fourth');
    });
  });
});
