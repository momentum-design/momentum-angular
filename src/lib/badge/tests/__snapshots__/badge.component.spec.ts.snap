// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BadgeComponent Classes and ngClass should pass classNames to the component 1`] = `
<ng-component>
  <md-badge
    class="third fourth md-badge first second md-badge--blue md-badge--round"
    color="blue"
    ng-reflect-color="blue"
    ng-reflect-klass="third fourth"
    ng-reflect-ng-class="first second"
    ng-reflect-rounded="true"
  >
    Testing Classes
  </md-badge>
</ng-component>
`;

exports[`BadgeComponent Default, Rounded and Colors should assign conditional color class name 1`] = `
<md-badge
  _color={[Function String]}
  elementRef={[Function ElementRef]}
  rounded="false"
>

</md-badge>
`;

exports[`BadgeComponent Default, Rounded and Colors should assign conditional rounded class name 1`] = `
<md-badge
  _color="null"
  elementRef={[Function ElementRef]}
  rounded={[Function Boolean]}
>

</md-badge>
`;

exports[`BadgeComponent Default, Rounded and Colors should create and match snapshot 1`] = `
<md-badge
  _color="null"
  elementRef={[Function ElementRef]}
  rounded="false"
>

</md-badge>
`;
